hudi-examples/hudi-examples-spark/src/test/java/org/apache/hudi/examples/quickstart/TestHoodieSparkQuickstart.java

```diff
@@ -100,15 +107,25 @@ public void testHoodieSparkQuickstart() {
     String tablePath = tablePath(tableName);
 
     try {
-      HoodieSparkQuickstart.insertData(spark, jsc, tablePath, tableName, DATA_GEN);
-      HoodieSparkQuickstart.updateData(spark, jsc, tablePath, tableName, DATA_GEN);
+      final HoodieExampleDataGenerator<HoodieAvroPayload> dataGen = new HoodieExampleDataGenerator<>();
+
+      insertData(spark, jsc, tablePath, tableName, dataGen);
+      queryData(spark, jsc, tablePath, tableName, dataGen);
+
+      updateData(spark, jsc, tablePath, tableName, dataGen);
+      queryData(spark, jsc, tablePath, tableName, dataGen);
+
+      incrementalQuery(spark, tablePath, tableName);
+      pointInTimeQuery(spark, tablePath, tableName);
+
+      delete(spark, tablePath, tableName);
+      queryData(spark, jsc, tablePath, tableName, dataGen);
 
-      HoodieSparkQuickstart.queryData(spark, jsc, tablePath, tableName, DATA_GEN);
-      HoodieSparkQuickstart.incrementalQuery(spark, tablePath, tableName);
-      HoodieSparkQuickstart.pointInTimeQuery(spark, tablePath, tableName);
+      insertOverwriteData(spark, jsc, tablePath, tableName, dataGen);
+      queryData(spark, jsc, tablePath, tableName, dataGen);
```

Comment:
nit: Looks the same as the main method of HoodieSparkQuickstart.  Could the same test flow as HoodieSparkQuickstart (insert, update, query, delete, etc.) be extracted? (can be a follow-up PR)

Reply:
Good call!